src/helperFunctions/install.py

```diff
@@ -3,17 +3,26 @@
 import os
 import shutil
 from pathlib import Path
-from typing import List, Union
+from typing import List, Tuple, Union
 
 from common_helper_process import execute_shell_command_get_return_code
 
 
 class InstallationError(Exception):
-    pass
+    '''
+    Class representing all expected errors that happen during installation, such as timeouts on remote hosts.
+    '''
 
 
 class OperateInDirectory:
-    def __init__(self, target_directory: Union[str, Path], remove=False):
+    '''
+    Context manager allowing to execute a number of commands in a given directory. On exit, the working directory is
+    restored to its previous state.
```

Comment:
IMHO this could be misunderstood as "changes inside the directory are reverted" when you actually mean that the current working directory is reset to the previous directory.

Reply:
I had this in mind already when writing this. I guess it's back to the drawing board...